### PR TITLE
Add test fixtures that allow running tests only for specific db backend

### LINKED ISSUE #4278: Add `pytest` fixture to allow marking tests that should only be run for a particular database backend
This will allow us to get rid of most of database backend specific folders in `tests/backens`

### PR DESCRIPTION
**Pull request by sphuber**
Fixes #4278 

The new fixtures `skip_if_not_sqlalchemy` and `skip_if_not_django` are
added. Any test using this fixture will be skipped if the loaded profile
does not use the SqlAlchemy or Django backend, respectively. This allows
to move some tests that were defined in `tests/backends/aiida_sqlalchemy`
to be moved to the backend agnostic location.

A few other tests that in the backend specific folder were removed since
they were already tested in the corresponding `tests/orm/implementation`
files.

### CODE REVIEW

File: tests/backends/aiida_django/test_generic.py [hunk 1]
```diff
@@ -1,134 +0,0 @@
...
-"""
-Generic tests that need the use of the DB
-"""
-
-from aiida import orm
-from aiida.backends.testbase import AiidaTestCase
-from aiida.orm import Data
-
-
-class TestComputer(AiidaTestCase):
```
Comment (ramirezfranciscof) [inline tests/backends/aiida_django/test_generic.py:19]: So, it seems these tests for computer deletion are no more. Were these already being tested elsewhere and this was redundant?

File: tests/backends/aiida_django/test_generic.py [hunk 2]
```diff
@@ -1,134 +0,0 @@
...
-        Test if queries are working
-        """
-        from aiida.common.exceptions import NotExistent, MultipleObjectsError
-
-        backend = self.backend
-
-        default_user = backend.users.create('{}@aiida.net'.format(self.id())).store()
-
-        g_1 = backend.groups.create(label='testquery1', user=default_user).store()
-        self.addCleanup(lambda: backend.groups.delete(g_1.id))
```
[the remark this answers is not in the document]
Comment (sphuber) [reply tests/backends/aiida_django/test_generic.py:63]: These tests are some of the oldest and this was the apparently the manner chosen to clean up the database after the test finishes. The `addCleanup` is inherited from `unittest.TestCase` and takes a callable which is called after the test scope finishes.

File: tests/orm/implementation/test_groups.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    n_1 = orm.Data().store().backend_entity
+    n_2 = orm.Data().store().backend_entity
+    n_3 = orm.Data().store().backend_entity
+    n_4 = orm.Data().store().backend_entity
+
+    g_1.add_nodes([n_1, n_2])
+    g_2.add_nodes([n_1, n_3])
+
+    # NOTE: Here we pass type_string to query and get calls so that these calls don't
+    # find the autogroups (otherwise the assertions will fail)
```
[the remark this answers is not in the document]
Comment (sphuber) [reply tests/orm/implementation/test_groups.py:35]: Don't know why it was moved up, but the whole note doesn't make sense. The autogrouping is a front end feature and so should not come in to play. We can simply remove it along with all the explicit definition of `type_string` in the queries

File: tests/backends/aiida_django/test_generic.py (hunk 1 above)
Comment (sphuber) [reply tests/backends/aiida_django/test_generic.py:19]: Besides the fact that this is intended to be a backend specific test, yet it only uses the front end classes (making it pointless), this is already tested in the `tests/orm/implementation` tests

File: aiida/manage/tests/__init__.py [hunk 4]
```diff
@@ -138,6 +138,7 @@ def __init__(self, profile_name):
         check_if_tests_can_run()
 
         self._select_db_test_case(backend=self._profile.database_backend)
+        os.environ['AIIDA_TEST_BACKEND'] = self._profile.database_backend
```
[the remark this answers is not in the document]
Comment (sphuber) [reply aiida/manage/tests/__init__.py]: This is a remnant of my first attempt. I wanted originally to just make a function that would return the loaded backend and use that with the `pytest.mark.skipif` decorators. However, these are executed at startup  time, at which point the test environment including the test profile has not yet been loaded, making it impossible to determine what the database backend is and therefore whether the test should be run. The fixture is executed only after the the profile is loaded, making it possible to know the backend.

**Pull request merged**